source/MaterialXGenOgsXml/OgsFragment.cpp

```diff
@@ -65,11 +65,10 @@ class GlslGeneratorWrapperBase
         // Use FIS environment lookup for surface shader generation but
         // disable for texture nodes to avoid additional unneeded XML parameter
         // generation.    
-        genOptions.hwSpecularEnvironmentMethod =
-            _isSurface ? mx::SPECULAR_ENVIRONMENT_FIS : mx::SPECULAR_ENVIRONMENT_NONE;
+        genOptions.hwSpecularEnvironmentMethod = mx::SPECULAR_ENVIRONMENT_NONE;
 
         // Set to use no direct lighting
-        genOptions.hwMaxActiveLightSources = 0;
+        genOptions.hwMaxActiveLightSources = _isSurface ? 16 : 0;
```

Comment:
Curious about this. I know the default number of lights is 16 but it can go higher. Does this need to be passed in as an argument?

Reply:
Maya switches from 16 to infinite lights based on usage of the "active lights" mode. It works by adding more lights to the main function as long as the compiler agrees to compile the shader. The light integration currently done depends on storing light values from the main function so they are available when the MaterialX light loop is computed later. There needs to be a finite number of memory slots.

I would say we leave it at 16 for the moment until we find a way to unify the Maya and MaterialX light loops. If this becomes too inconvenient for production scenes, then we could make this an argument.